### ISSUE TITLE
CPU running at 100% caused by theme check

### ISSUE THREAD

**Issue by ComicDansMS** [Shopify/theme-check-vscode#78]
There is some sort of issue where one core of the CPU will start running at 100%, VS Code source control won't stop loading (not able to commit when this happens), and other search functions (ctrl+P) are slow or stop finding what you're looking for. I close VS Code to fix it but it will happen again in a few minutes.

I've found it is this extension due to it never happening when I'm not working on Shopify/liquid files, and disabling all related plugins except this one.

It happens across both my Macbook and my Linux (Manjaro) machine.

I'd love to give more information but this is as much as I can work out at the moment. It's been happening for at least a few weeks.

**Comment by charlespwd** [comment]
Can you upgrade your Shopify CLI and try again? The latest version of theme-check is 1.10+

I have a vague feeling that we might have fixed this already.

**Comment by ComicDansMS** [comment]
Sorry for the late response.,

I upgraded Shopify CLI and I've found the problem still persists on both machines (macbook and linux/manjaro).

On the macbook, `shopify theme check --version` is currently 1.10.2.

**Comment by ComicDansMS** [comment]
Yes I believe VS Code is built on electron. The macbook was reporting it to be electron, where my Linux machine is reporting it to be Code.

**- Does it happen every time you open a theme repo?** I'd say it happens almost everytime, if not everytime.
**- Is there a .theme-check.yml file in there?** Yes
**- Does it happen as soon as you open the theme or after a little while?** It doesn't happen immediatey, but probably within ten minutes or so.
**- Is there anything about your workflow that you think is unconventional that I should be aware of?** Not really. I fire up `shopify theme serve` in a terminal/console window and start working. 

I'll try to keep an eye on it and update. I'm not doing as much Shopify development at this point in time so sometimes there are periods where I forget about this issue that's open.

**Comment by ComicDansMS** [comment]
I've just setup a new macbook and found this to carry over.
Not sure if this would help at all, but this is some info from terminal on the process that's getting stuck..

`3:33.36 /Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper.app/Contents/MacOS/Code Helper --ms-enable-electron-run-as-node --inspect-port=0 /Applications/Visual Studio Code.app/Contents/Resources/app/out/bootstrap-fork --type=extensionHost --skipWorkspaceStorageLock`
